### ISSUE TITLE
can UnreahableCode rule catch code after function calls returning `Nothing`?

### ISSUE THREAD

**Issue by noamfreeman** [detekt/detekt#4606]
## Expected Behavior of the rule
while the UnreachableCode rule catches many cases,
if doesn't catch code occurring after a call to a function returning nothing.
for example, while jetbrains IDE's inspections catch the `return` line here, as unreachable
```kotlin
fun a(): Int {
    error("")
    return 2
}
``` 

Detekt doesn't warn about this case.
(errors definition is `public inline fun error(message: Any): Nothing = throw IllegalStateException(message.toString())`, returning `Nothing`)

## Context
in my project, i use Detekt to lint for bugs on CI,
 and don't rely heavily on compiler warnings (hard to work with in development, hard to configure as a separate short task as Detekt, since it's caching results)
 or IDE inspections (Detekt seems faster, easier to config, changes less, and has clearer results)

in one case, when a developer was trying to manually generate rare&complicated exception cases
 some code like the example above, managed to slip through review/tests/Detekt, while both compiler-warning or IDE-inspections would have catched it immediately.

**Comment by cortinico** [comment]
> while jetbrains IDE's inspections catch the `return` line here, as unreachable

We should look into seeing if this is coming from a compiler warning or a IDE inspection and potentially wrap it.

**Comment by BraisGabin** [comment]
Are you running detekt with type solving enabled?

The logic of this rule is completely delegated to the compiler so it should work out of the box already.

https://github.com/detekt/detekt/blob/f32d0617336dbf54fb81cc6129a8df58e3a383e7/detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UnreachableCode.kt#L49-L59

**Comment by noamfreeman** [comment]
thanks @BraisGabin . i'm using gradle, and seems that running `detektDebug` instead of `detekt` does catch it.
